src/Graphql/Type/SchemaBuilder.php

```diff
@@ -40,47 +42,49 @@
  */
 final class SchemaBuilder implements SchemaBuilderInterface
 {
+    use ClassInfoTrait;
+
     private $propertyNameCollectionFactory;
     private $propertyMetadataFactory;
     private $resourceNameCollectionFactory;
     private $resourceMetadataFactory;
     private $collectionResolverFactory;
-    private $itemResolverFactory;
+    private $itemResolver;
     private $itemMutationResolverFactory;
     private $defaultFieldResolver;
     private $paginationEnabled;
-    private $resourceTypesCache = [];
+    private $graphqlTypes = [];
 
-    public function __construct(PropertyNameCollectionFactoryInterface $propertyNameCollectionFactory, PropertyMetadataFactoryInterface $propertyMetadataFactory, ResourceNameCollectionFactoryInterface $resourceNameCollectionFactory, ResourceMetadataFactoryInterface $resourceMetadataFactory, ResolverFactoryInterface $collectionResolverFactory, ResolverFactoryInterface $itemResolverFactory, ResolverFactoryInterface $itemMutationResolverFactory, callable $defaultFieldResolver, bool $paginationEnabled = true)
+    public function __construct(PropertyNameCollectionFactoryInterface $propertyNameCollectionFactory, PropertyMetadataFactoryInterface $propertyMetadataFactory, ResourceNameCollectionFactoryInterface $resourceNameCollectionFactory, ResourceMetadataFactoryInterface $resourceMetadataFactory, ResolverFactoryInterface $collectionResolverFactory, callable $itemResolver, ResolverFactoryInterface $itemMutationResolverFactory, callable $defaultFieldResolver, bool $paginationEnabled = true)
     {
         $this->propertyNameCollectionFactory = $propertyNameCollectionFactory;
         $this->propertyMetadataFactory = $propertyMetadataFactory;
         $this->resourceNameCollectionFactory = $resourceNameCollectionFactory;
         $this->resourceMetadataFactory = $resourceMetadataFactory;
         $this->collectionResolverFactory = $collectionResolverFactory;
-        $this->itemResolverFactory = $itemResolverFactory;
+        $this->itemResolver = $itemResolver;
         $this->itemMutationResolverFactory = $itemMutationResolverFactory;
         $this->defaultFieldResolver = $defaultFieldResolver;
         $this->paginationEnabled = $paginationEnabled;
     }
 
     public function getSchema(): Schema
     {
-        $queryFields = [];
+        $nodeInterface = $this->getNodeInterface();
```

Comment:
Why not storing the type in `$graphqlTypes` instead of passing it in all the methods?